prov/efa/src/efa_domain.c

```diff
@@ -76,19 +76,19 @@ static int efa_domain_init_device_and_pd(struct efa_domain *efa_domain,
 	if (!domain_name)
 		return -FI_EINVAL;
 
-	for (i = 0; i < g_device_cnt; i++) {
-		device_name = g_device_list[i].ibv_ctx->device->name;
+	for (i = 0; i < g_efa_selected_device_cnt; i++) {
```

Comment:
Have to say, I don't understand this change.  Why would you ever care about the discovered device count length?  and with this change, you've made the patch way bigger.

Reply:
We need the count to iterate over the discovered devices

I renamed the variables because I felt the names `g_device_cnt` and `g_device_list` were ambiguous re whether they were all IB devices or EFA devices or the selected EFA devices that match FI_EFA_IFACE